dictionaries/CallMap_80_delta.php

```diff
@@ -498,6 +498,10 @@
 'read_exif_data' => ['array', 'filename'=>'string', 'sections_needed='=>'string', 'sub_arrays='=>'bool', 'read_thumbnail='=>'bool'],
 'ReflectionClass::getConstants' => ['array<string,mixed>'],
 'ReflectionType::isBuiltin' => ['bool'],
+'ReflectionFunction::getReturnType' => ['ReflectionNamedType|ReflectionUnionType|null'],
+'ReflectionFunctionAbstract::getReturnType' => ['ReflectionNamedType|ReflectionUnionType|null'],
+'ReflectionMethod::getReturnType' => ['ReflectionNamedType|ReflectionUnionType|null'],
+'ReflectionParameter::getType' => ['ReflectionNamedType|ReflectionUnionType|null'],
```

Comment:
I think we should use `ReflectionType` here:

```suggestion
'ReflectionFunction::getReturnType' => ['ReflectionType|null'],
'ReflectionFunctionAbstract::getReturnType' => ['ReflectionType|null'],
'ReflectionMethod::getReturnType' => ['ReflectionType|null'],
'ReflectionParameter::getType' => ['ReflectionType|null'],
```

This should also fix the two Psalm errors.

Reply:
it depends on what we are going to archive:
 a) specify all more specific types
 b) specify fewer types by using parent type

I assume the goal of the psalm is to have as much as possible types information, this is why a) is preferable IMO.

Just for a reference: 
- `ReflectionType` has 2 child classes:
  - `ReflectionUnionType` (methods: `getTypes`)
  - `ReflectionNamedType` (methods: `getName`, `isBuiltin`)

Reply:
just created a snippet to test all possible cases for PHP 8.0: https://3v4l.org/QE9Yn

Reply:
> I assume the goal of psalm is to have as much as possible types information, this is why a) is preferable IMO.

I get your point, but even when using `ReflectionNamedType|ReflectionUnionType` over `ReflectionType`, you will still need to check the type with `instanceof` first, before you can call any of those three methods you mentioned.

Essentially this is narrowing the type more than the PHP's public API promises, and would prevent Psalm users from handling the case that another `ReflectionType` was returned. On the other hand, this could also be considered a nice side-effect.

My guess is that Psalm tries to stay close to the official PHP signatures, but maybe this is the right moment to ask @muglug or @weirdan for their opinion?